latch/account.py

```diff
@@ -1,45 +1,95 @@
-from dataclasses import dataclass
-from typing import List, Optional
+from dataclasses import dataclass, field
+from functools import cache
+from typing import List, Literal, Optional, TypedDict, overload
 
-from latch.gql.execute import execute
+import gql
+
+from latch.gql._execute import execute
 from latch.registry.project import Project
+from latch_cli.config.user import user_config
+
+
+class _CatalogProjectNode(TypedDict):
+    id: str
+    displayName: str
+
+
+@dataclass
+class _Cache:
+    catalog_projects: Optional[List[Project]] = None
 
 
 @dataclass(frozen=True)
 class Account:
+    _cache: _Cache = field(
+        default_factory=lambda: _Cache(),
+        init=False,
+        repr=False,
+        hash=False,
+        compare=False,
+    )
+
     id: str
 
     @classmethod
+    @cache
     def current(cls):
-        account_id = execute(
-            document="""
-                query accountInfoQuery {
-                    accountInfoCurrent {
-                        id
+        if user_config.workspace != "":
+            account_id = user_config.workspace
+        else:
+            account_id = execute(
+                document=gql.gql("""
+                    query accountInfoQuery {
+                        accountInfoCurrent {
+                            id
+                        }
                     }
-                }
-            """,
-        )["accountInfoCurrent"]["id"]
+                    """),
+            )["accountInfoCurrent"]["id"]
 
         return cls(id=account_id)
 
-    def list_projects(self) -> List[Project]:
-        query = f"""
-            query ProjectsQuery ($argOwnerId: BigInt!) {{
-                catalogProjects (
-                    condition: {{
-                        ownerId: $argOwnerId
-                        removed: false
-                    }}
-                ) {{
-                    nodes {{
-                        id
-                        displayName
-                    }}
-                }}
-            }}
-        """
-
-        data = execute(query, {"argOwnerId": self.id})
-
-        return [Project(node["id"]) for node in data["catalogProjects"]["nodes"]]
+    def load(self):
+        data = execute(
+            gql.gql("""
+            query AccountQuery($ownerId: BigInt!) {
+                accountInfo(id: $ownerId) {
+                    catalogProjectsByOwnerId(
+                        condition: {
+                            removed: false
+                        }
+                    ) {
+                        nodes {
+                            id
+                            displayName
+                        }
+                    }
+                }
+            }
+            """),
+            {"ownerId": self.id},
+        )["accountInfo"]
+        # todo(maximsmol): deal with nonexistent accounts
```

Comment:
I would load in the experiments as well. The query will still be small and fast.